# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/CoalescedInputData.h
# Commit: f8a2a169b296
# Full Hash: f8a2a169b2969d29e04733ae27a56459e5359973
# Author: Alexandru Marc <amarc@mozila.com>
# Date: 2025-05-16 09:19:19
# Regressor Bug: 1550462
# File Overlap Count: 1
# Description:
#   Revert "Bug 1550462 - part 3: Enable `pointerrawupdate` event r=smaug" for causing mochitest failures @ test_pointerrawupdate_event_count.html
#   
#   This reverts commit eb27445e0352d6b4df540adb40e21f72fc052e80.
#   
#   Revert "Bug 1550462 - part 2: Make `PresShell::HandleEvent` dispatch preceding `pointerrawupdate` event r=smaug,dom-core,edgar"
# ==============================================================================

diff -r 8d25773fc82f -r f8a2a169b296 dom/ipc/CoalescedInputData.h
--- a/dom/ipc/CoalescedInputData.h	Fri May 16 06:20:47 2025 +0000
+++ b/dom/ipc/CoalescedInputData.h	Fri May 16 09:30:09 2025 +0300
@@ -24,21 +24,12 @@
 
   UniquePtr<InputEventType> mCoalescedInputEvent;
   ScrollableLayerGuid mGuid;
-  uint64_t mInputBlockId = 0;
-  uint32_t mGeneration = 0;
-
-  void AdvanceGeneration() {
-    if (!IsEmpty()) {
-      mGeneration++;
-    }
-  }
+  uint64_t mInputBlockId;
 
  public:
-  CoalescedInputData() = default;
+  CoalescedInputData() : mInputBlockId(0) {}
 
   void RetrieveDataFrom(CoalescedInputData& aSource) {
-    aSource.AdvanceGeneration();
-    AdvanceGeneration();
     mCoalescedInputEvent = std::move(aSource.mCoalescedInputEvent);
     mGuid = aSource.mGuid;
     mInputBlockId = aSource.mInputBlockId;
@@ -51,23 +42,12 @@
                    const uint64_t& aInputBlockId);
 
   UniquePtr<InputEventType> TakeCoalescedEvent() {
-    AdvanceGeneration();
     return std::move(mCoalescedInputEvent);
   }
 
   ScrollableLayerGuid GetScrollableLayerGuid() { return mGuid; }
 
   uint64_t GetInputBlockId() { return mInputBlockId; }
-
-  /**
-   * The generation number of the latest state stored by the instance.
-   * It'll be incremented when the coalesced event data is retrieved or taken.
-   * So, this is useful to avoid handling same coalesced events twice when
-   * a nested event loop may handle this.
-   * NOTE: You should compare the value only with `==` or `!=`.  Do not use
-   * `<` nor `>` because the value may circulate to 0 from UINT32_MAX.
-   */
-  [[nodiscard]] uint32_t Generation() const { return mGeneration; }
 };
 
 class CoalescedInputFlusher : public nsARefreshObserver {
@@ -81,17 +61,12 @@
   void StartObserver();
   void RemoveObserver();
 
-  /**
-   * Return a refresh driver which is proper one for BrowserChild.
-   * Note that this is not a getter of mRefreshDriver.
-   */
-  [[nodiscard]] nsRefreshDriver* GetRefreshDriver();
-
  protected:
   virtual ~CoalescedInputFlusher();
 
+  nsRefreshDriver* GetRefreshDriver();
+
   BrowserChild* mBrowserChild;
-  // A refresh driver which this instance waits for the next refresh of.
   RefPtr<nsRefreshDriver> mRefreshDriver;
 };
 }  // namespace mozilla::dom